xds/internal/server/listener_wrapper.go

```diff
@@ -297,11 +301,22 @@ func (l *listenerWrapper) Accept() (net.Conn, error) {
 			conn.Close()
 			continue
 		}
-		// TODO: once matched an accepted connection to a filter chain,
-		// instantiate the HTTP filters in the filter chain + the filter
-		// overrides, pipe filters and route into connection, which will
-		// eventually be passed to xdsUnary/Stream interceptors.
-		return &connWrapper{Conn: conn, filterChain: fc, parent: l}, nil
+		var rc xdsclient.RouteConfigUpdate
+		if fc.InlineRouteConfig != nil {
+			rc = *fc.InlineRouteConfig
+		} else {
+			rcPtr := l.rdsUpdates
+			rcuPtr := (*map[string]xdsclient.RouteConfigUpdate)(rcPtr)
+			rcu := *rcuPtr
+			rc = rcu[fc.RouteConfigName]
+		}
+
+		if err := fc.ConstructUsableRouteConfiguration(rc); err != nil {
```

Comment:
The call to `ConstructUsableRouteConfiguration` actually updates the `filterChain` receiver on which it is called. Is this supposed to be a one-time operation? If so, how do you guarantee that? If not, how do you guarantee that concurrent calls to it do not stomp on each other?

Reply:
No, this isn't guaranteed to be a one time operation. How can this be called concurrently? Filter Chain configuration is tied to a Listener, and thus the xdsclient constructs a FilterChainManager for a specific Listener. Is there a possibility for two listeners to use the same FCM and thus map to another? I don't think it's possible to accept two connections concurrently, due to the for { wait }? Or I am understanding this wrong?

Reply:
It would be good to document this whole decision process somewhere. A code comment is totally fine. We want someone reading the code to understand the tradeoffs that we have made and the reasoning behind them. Thanks.

Reply:
Added a comment explaining thought process.

Reply:
Is this correct, though?

You are updating `Interceptors` in `fc`, but `fc` is shared between multiple connections.  I don't see a mutex or atomic here or anything, either.

Reply:
Sure, fc is shared between multiple connections, but Interceptors is only read/written in this accept() while true loop, which will never have two things updating concurrently?

Reply:
`fc` is a local pointer to shared memory, is it not?  If multiple listeners are referencing the same memory then you have a race.

Reply:
How can multiple Listeners reference the same fc? The fc is a data structure built during an LDS update, which is tied to a listener wrapper. Thus, the fc data structure is not shared amongst listeners, but is PER listener, correct?

Reply:
If that's the case, it's honestly hard to tell from looking at the code here.  `fc` is `l.filterChains.Lookup`.  `l.filterChains` comes from `switchMode` which is presumably usually called via `handleLDSUpdate` with the parameter `ilc.FilterChains`.  `ilc` is the LDS update's `update.InboundListenerConfig` which appears to be constructed once per xds update, not once per listener.

Reply:
Sent out a separate PR addressing this comment.